src/status_im/extensions/capacities/events.cljs

```diff
@@ -0,0 +1,687 @@
+(ns status-im.extensions.capacities.events
+  (:require [re-frame.core :as re-frame]
+            [status-im.utils.handlers :as handlers]
+            [clojure.string :as string]
+            [status-im.chat.commands.sending :as commands-sending]
+            [status-im.ui.screens.navigation :as navigation]
+            [status-im.ipfs.core :as ipfs]
+            [status-im.browser.core :as browser]
+            [status-im.utils.fx :as fx]
+            [status-im.accounts.update.core :as accounts.update]
+            status-im.extensions.capacities.ethereum
+            status-im.extensions.capacities.camera.events
+            status-im.extensions.capacities.network))
+
+(defn- empty-value? [o]
+  (cond
+    (seqable? o) (empty? o)
+    :else (nil? o)))
+
+(defn- put-or-dissoc [db id key value]
+  (if (empty-value? value)
+    (update-in db [:extensions/store id] dissoc key)
+    (assoc-in db [:extensions/store id key] value)))
+
+(defn- append [acc k v]
+  (let [o (get acc k)
+        ve (cond (vector? o) o
+                 (not (nil? o)) (vector o)
+                 :else [])]
+    (assoc acc k (conj ve v))))
+
+(defn- json? [res]
+  (when-let [type (get-in res [:headers "content-type"])]
+    (string/starts-with? type "application/json")))
+
+(defn- parse-json [o]
+  (when o
+    (js->clj (js/JSON.parse o) :keywordize-keys true)))
+
+(defn- parse-result [o on-success]
+  (let [res (if (json? o) (update o :body parse-json) o)]
+    (on-success res)))
+
+(defn operation->fn [k]
+  (case k
+    :plus +
+    :minus -
+    :times *
+    :divide /))
+
+(defn update-account-data [{db :db :as cofx} id data-fn]
+  (let [{:account/keys [account]} db
+        extensions     (get account :extensions)]
+    (accounts.update/account-update
+     cofx
+     {:extensions (update-in extensions [id :data] data-fn)}
+     {})))
+
+(defn- put-or-dissoc-persis [current-data key value]
```

Comment:
Just a suggestion. Instead of creating new fn `put-or-dissoc-persis`, you can utilize clojure multi-arity support here so`put-or-dissoc` may have two bodies for 3 and 4 arguments.